libraries/grpc-sdk/src/classes/ModuleManager.ts

```diff
@@ -22,38 +22,41 @@ export class ModuleManager {
   }
 
   start() {
-    this.module.initialize(this.grpcSdk);
-    const self = this;
-    this.preRegisterLifecycle()
-      .then(() => {
-        const url = (process.env.REGISTER_NAME === 'true'
-          ? `${self.module.name}:`
-          : `${self.serviceAddress}:`) + self.module.port;
-        return self.grpcSdk.config.registerModule(self.module.name, url);
-      })
-      .catch((err: Error) => {
-        console.log('Failed to initialize server');
-        console.error(err);
-        process.exit(-1);
-      })
-      .then(() => {
-        return this.postRegisterLifecycle();
-      })
-      .catch((err: Error) => {
-        console.log('Failed to activate module');
-        console.error(err);
-      });
+    this.grpcSdk.initialize().then(() => {
```

Comment:
I reeeally dislike seeing these then blocks, isn't there a better way to write this?

Reply:
I guess as `ModuleManager.start()` is the last thing to be called before module initialization takes over in `index.ts` files we could just turn the whole thing async without even requiring top-level await.